cmd/cluster/transferowner.go

```diff
@@ -54,9 +66,175 @@ func newTransferOwnerOptions(streams genericclioptions.IOStreams, globalOpts *gl
+	if err != nil {
+		return err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: hiveNamespace,
+		},
+		Type: corev1.SecretTypeDockerConfigJson,
+		Data: map[string][]byte{
+			".dockerconfigjson": pullsecret,
+		},
+	}
+	_, err = clientset.CoreV1().Secrets(hiveNamespace).Create(context.TODO(), secret, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	//Create a syncset on hive to push the pull secret down
+	syncsetName := "pull-secret-replacement"
+
+	err = createSyncSet(syncsetName, hiveNamespace, cdName, kubeCli)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func createSyncSet(syncSetName string, hiveNamespace string, cdName string, kubeCli client.Client) error {
+	ctx := context.TODO()
+
+	syncSet := &hiveapiv1.SyncSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      syncSetName,
+			Namespace: hiveNamespace,
+		},
+		Spec: hiveapiv1.SyncSetSpec{
+			ClusterDeploymentRefs: []corev1.LocalObjectReference{
+				{
+					Name: cdName,
+				},
+			},
+			SyncSetCommonSpec: hiveapiv1.SyncSetCommonSpec{
+				ResourceApplyMode: "Upsert",
+				Secrets: []hiveapiv1.SecretMapping{
+					{
+						SourceRef: hiveapiv1.SecretReference{
+							Name:      "pull",
+							Namespace: hiveNamespace,
+						},
+						TargetRef: hiveapiv1.SecretReference{
+							Name:      "pull-secret",
+							Namespace: "openshift-config",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	err := kubeCli.Create(ctx, syncSet)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("SyncSet %s in namespace %s has been created.\n", syncSetName, hiveNamespace)
+
+	hiveinternalv1alpha1.AddToScheme(kubeCli.Scheme())
+	searchStatus := &hiveinternalv1alpha1.ClusterSync{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cdName,
+			Namespace: hiveNamespace,
+		},
+	}
+	foundStatus := &hiveinternalv1alpha1.ClusterSync{}
+	isSSSynced := false
+	for i := 0; i < 24; i++ {
+		err = kubeCli.Get(ctx, client.ObjectKeyFromObject(searchStatus), foundStatus)
+		if err != nil {
+			return err
+		}
+
+		for _, status := range foundStatus.Status.SyncSets {
+			if status.Name == syncSetName {
+				if status.FirstSuccessTime != nil {
+					isSSSynced = true
+					break
+				}
+			}
+		}
+
+		if isSSSynced {
+			fmt.Printf("\nSync completed...\n")
+			break
+		}
+
+		fmt.Printf(".")
+		time.Sleep(time.Second * 5)
+	}
+	if !isSSSynced {
+		return fmt.Errorf("syncset failed to sync. Please verify syncset is still there and manually delete syncset ")
+	}
+
+	// Clean up the SS on hive
+	err = kubeCli.Delete(ctx, syncSet)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func rolloutTelemeterClientPods(clientset *kubernetes.Clientset, namespace, selector string) error {
+	// Delete pods with the specified label selector in the specified namespace.
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: selector,
+	})
+	if err != nil {
+		return err
```

Comment:
There's a few more of these, I'll leave it to you to determine what you think is the best error message to accompany 😄